### PR TITLE
Added correct location of SMT repositories for Cloud6

### PR DESCRIPTION
**Pull request by jsuchome**
Requires https://github.com/crowbar/crowbar-core/pull/125

### CODE REVIEW

**Comment by jsuchome** [comment]
@vuntz ?

File: src/modules/Crowbar.rb [hunk 1]
```diff
@@ -58,6 +58,20 @@ def main
...
+        "sles12-updates"                => "repo/SUSE/Updates/SLE-SERVER/12/%s/update",
+        "sles12-sp1-pool"               => "repo/SUSE/Products/SLE-SERVER/12-SP1/%s/product",
+        "sles12-sp1-updates"            => "repo/SUSE/Updates/SLE-SERVER/12-SP1/%s/update",
+        "suse-openstack-cloud-6-pool"   => "repo/SUSE/Products/OpenStack-Cloud/6/%s/product",
+        "suse-openstack-cloud-6-updates" => "repo/SUSE/Updates/OpenStack-Cloud/6/%s/update",
+        "sle12-sp1-ha-pool"             => "repo/SUSE/Products/SLE-HA/12-SP1/%s/product",
+        "sle12-sp1-ha-updates"          => "repo/SUSE/Updates/SLE-HA/12-SP1/%s/update",
+        "suse-enterprise-storage-2-pool" => "repo/SUSE/Products/Storage/2/%s/product",
+        "suse-enterprise-storage-2-updates" => "repo/SUSE/Updates/Storage/2/%s/update"
+      }
```
Comment (vuntz) [inline src/modules/Crowbar.rb]: We can put that info in the yaml file; that would make it possible to not have yast know about anything from the repo. What do you think?

**Comment by vuntz** [comment]
Two other comments:

- we also need to update `InitRPRepos` for the detection of SMT or not

- there's an example of SMT path (http://smt.example.com/repo/$RCE) that needs an update

File: src/modules/Crowbar.rb (hunk 1 above)
Comment (kobliha) [inline src/modules/Crowbar.rb]: From my POV, this is a configuration that definitely belongs to a configuration file. It's not a good idea to have it within the source code.

Comment (jsuchome) [inline src/modules/Crowbar.rb]: Right, much better idea

Comment (jsuchome) [inline src/modules/Crowbar.rb]: https://github.com/crowbar/crowbar-core/pull/125

File: src/include/crowbar/complex.rb [hunk 2]
```diff
@@ -645,11 +645,15 @@ def HandleLocationURL(key, event)
...
 
+      smt_repos = Crowbar.smt_repos
+
       @repos.each do |platform, arches|
         arches.each do |arch, repos|
           distro = platform == "suse-12.1" ? "sles12-sp1-#{arch}" : "sles12-#{arch}"
           repos.each do |repo_name, r|
-            url = "#{@remote_server_url}/repo/$RCE/#{repo_name}/#{arch}/"
+            if @repos_location == "smt"
+              url = "#{@remote_server_url}/repo/#{@repos[platform][arch][repo_name]['smt_path'] || ''}"
```
Comment (vuntz) [inline src/include/crowbar/complex.rb]: I would actually not change the URL if there's no smt_path attribute.

Comment (jsuchome) [inline src/include/crowbar/complex.rb]: Oh, true, it should stay empty

**Comment by vuntz** [comment]
+1

File: src/include/crowbar/complex.rb [hunk 3]
```diff
@@ -645,16 +645,19 @@ def HandleLocationURL(key, event)
 
       return nil if @remote_server_url.empty? || @repos_location == "custom"
 
+      smt_repos = Crowbar.smt_repos
```
Comment (vuntz) [inline src/include/crowbar/complex.rb]: Can be dropped.

File: src/modules/Crowbar.rb [hunk 4]
```diff
@@ -320,6 +320,7 @@ def Write
 
     publish :variable => :network_file, :type => "string"
     publish :variable => :repos, :type => "map <string, map>"
+    publish :variable => :smt_repos, :type => "map <string, string>"
```
Comment (vuntz) [inline src/modules/Crowbar.rb]: Same here.

**Pull request merged**